### PR TITLE
Add OIDC documentation

### PR DESCRIPTION
**Pull request by fflorent**
Document how to use OpenID Connection with some example.

Most of this documentation has been inspired from the SAML connection documentation.

### CODE REVIEW

File: help/install/oidc.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+====
+
+Configuration for OIDC, useful for organization single-sign-on logins.
+A good informative overview of OIDC is at <https://developer.okta.com/blog/2019/10/21/illustrated-guide-to-oauth-and-oidc>
+
+Note:
+
+  * SP is "Service Provider", in our case, the Grist application.
+  * IdP is the "Identity Provider", somewhere users log into, e.g. Keycloak, Authelia, ...
+  * OIDC is the accronym for OpenID Connect
```
Comment (georgegevoian) [inline help/install/oidc.md:11]:
```suggestion
  * OIDC is the acronym for OpenID Connect
```

File: help/install/oidc.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  * `GRIST_OIDC_IDP_SKIP_END_SESSION_ENDPOINT` (optional) -  If set to "true", on logout, there won't be
+    any attempt to call the IdP's `end_session_endpoint` (the user will remain logged in in the IdP).
+    You should only set it to "true" if the IdP does not provide such endpoint (for example if you use Gitlab).
+  * `GRIST_OIDC_SP_PROFILE_NAME_ATTR` (optional) - The key of the attribute to use for the user's name. If omitted,
+    the name will be the concatenation of `given_name` + `family_name` if they are provided or the `name` attribute otherwise.
+  * `GRIST_OIDC_SP_PROFILE_EMAIL_ATTR` (optional) - The key of the attribute to use for the user's email. Defaults to "email".
+
+## Example: Gitlab
+
+See [how to create an OAuth2 application in Gitlab in this documentation](https://docs.gitlab.com/ee/integration/oauth_provider.html). While creating the application, set the redirection URI to `https://<grist-domain>/oauth2/callback` (or `http://localhost:8484/oauth2/callback` if tested locally, change `8484` to the port you listen on) and select the scopes you will specify in `GRIST_OIDC_IDP_SCOPES`.
```
Comment (georgegevoian) [inline help/install/oidc.md:33]:
```suggestion
See [how to create an OAuth2 application in Gitlab in this documentation](https://docs.gitlab.com/ee/integration/oauth_provider.html). While creating the application, set the redirect URI to `https://<grist-domain>/oauth2/callback` (or `http://localhost:8484/oauth2/callback` if tested locally, change `8484` to the port you listen on) and select the scopes you will specify in `GRIST_OIDC_IDP_SCOPES`.
```

File: help/install/oidc.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Once the application is set up and start grist with these settings:
```
Comment (georgegevoian) [inline help/install/oidc.md:35]:
```suggestion
Once the application is set up, start Grist with these settings:
```

File: help/install/oidc.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ - `GRIST_OIDC_SP_HOST` = `https://<grist-domain>` / `http://localhost:8484`
+ - `GRIST_OIDC_IDP_ISSUER` = `https://gitlab.com/.well-known/openid-configuration`
+ - `GRIST_OIDC_IDP_SCOPES` = `"openid profile email"`
+ - `GRIST_OIDC_IDP_CLIENT_ID` = "..." (the client ID generated by Gitlab for the application)
+ - `GRIST_OIDC_IDP_CLIENT_SECRET` = "..." (the client secret generated by Gitlab for the application)
+ - `GRIST_OIDC_IDP_SKIP_END_SESSION_ENDPOINT` = `true` (Gitlab don't propose `end_session_endpoint`)
+
+## Example: Auth0
+
+Create an application in Auth0 [as explained in this documentation](https://auth0.com/docs/get-started/auth0-overview/create-applications) (you can select the app type named `Regular Web Applications`). Once the application created, ensure to add at least the following configuration for the app:
```
Comment (georgegevoian) [inline help/install/oidc.md:46]:
```suggestion
Create an application in Auth0 [as explained in this documentation](https://auth0.com/docs/get-started/auth0-overview/create-applications) (you can select the app type named `Regular Web Applications`). Once the application is created, ensure to add at least the following configuration for the app:
```

File: help/install/oidc.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * Allowed callback URLs: `https://<grist-domain>/oauth2/callback`
+ * Allowed logout URLs: `http://<grist-domain>/*` (you can also replace the wildcard with the whole path in order to be stricter)
+
+Then you should be able to start Grist with the following settings:
+
+ - `GRIST_OIDC_SP_HOST` = `https://<grist-domain>` / `http://localhost:8484`
+ - `GRIST_OIDC_IDP_ISSUER` = `https://<your-auth0-app-domain>`
+ - `GRIST_OIDC_IDP_SCOPES` = `"openid profile email"`
+ - `GRIST_OIDC_IDP_CLIENT_ID` = "..." (the client ID generated by Auth0 for the application)
+ - `GRIST_OIDC_IDP_CLIENT_SECRET` = "..." (the client secret generated by Auht0 for the application)
```
Comment (georgegevoian) [inline help/install/oidc.md:57]:
```suggestion
 - `GRIST_OIDC_IDP_CLIENT_SECRET` = "..." (the client secret generated by Auth0 for the application)
```

File: help/install/oidc.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Example: Keycloak
+
+First of all, setup Keycloak as explained in one of these "Getting started" guides: <https://www.keycloak.org/guides#getting-started>.
```
Comment (georgegevoian) [inline help/install/oidc.md:61]:
```suggestion
First of all, set up Keycloak as explained in one of these "Getting started" guides: <https://www.keycloak.org/guides#getting-started>.
```

File: help/install/oidc.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+Once keycloak is setup, with a realm and a user, create a new client with the following configuration:
```
Comment (georgegevoian) [inline help/install/oidc.md:63]:
```suggestion
Once keycloak is set up with a realm and a user, create a new client with the following configuration:
```

File: help/install/oidc.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+ - Client type: OpenID Connect
+ - Authentication flow: Standard Flow
+ - Root URL: `https://<grist-domain>`
+ - Valid redirect URIs: `/oauth2/callback`
+ - Valid post logout redirect URIs: `/*` (you can also replace the wildcard with the whole path in order to be stricter)
+
+Submit your settings and go to the Credentials tab to retrieve the client Secret.
```
Comment (georgegevoian) [inline help/install/oidc.md:71]:
```suggestion
Submit your settings and go to the Credentials tab to retrieve the client secret.
```

File: help/install/oidc.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+Then, you can start Grist with the following configuration:
+
+ - `GRIST_OIDC_SP_HOST` = `https://<grist-domain>` / `http://localhost:8484`
+ - `GRIST_OIDC_IDP_ISSUER` = `https://<keycloak-domain>/realms/<your-realm>`
+ - `GRIST_OIDC_IDP_SCOPES` = `"openid profile email"`
+ - `GRIST_OIDC_IDP_CLIENT_ID` = "..." (the ID you chose for the Keycloak client)
+ - `GRIST_OIDC_IDP_CLIENT_SECRET` = "..." (the client secret generated by Keycload retrieved earlier)
```
Comment (georgegevoian) [inline help/install/oidc.md:79]:
```suggestion
 - `GRIST_OIDC_IDP_CLIENT_SECRET` = "..." (the client secret generated by Keycloak retrieved earlier)
```

**Review by georgegevoian — COMMENTED** [review]
Thanks @fflorent. Left some small suggestions.

**Comment by fflorent** [comment]
Thanks @georgegevoian!

**Review by georgegevoian — APPROVED** [review]
Thanks!

**Pull request merged**